scripts/rtl872x.tcl

```diff
@@ -87,15 +87,16 @@ set rtl872x_ready                       0
 proc rtl872x_fixup_address {address} {
     global FLASH_MMU_START_ADDR
     if {$address >= $FLASH_MMU_START_ADDR} {
-        return [expr $address - $FLASH_MMU_START_ADDR]
+        return [expr {$address - $FLASH_MMU_START_ADDR}]
     }
     return $address
 }
 
 proc openocd_read_register {reg} {
     set value ""
-    mem2array value 32 $reg 1
-    return $value(0)
+    set value [read_memory $reg 32 1]
```

Comment:
I'm pretty sure this one is going to fail. Because the way we're handling the value object in the follow lines.
Now the returned object is pretty different from the deprecated function